### ISSUE TITLE
Text with BOM marks is shown incorrectly in viewer and editor

### ISSUE THREAD

**Issue by e673** [elfmz/far2l#850]
When BOM mark appears in the middle of the text (i.e. in log files), an extra space is added after the next word when using 65001 encoding.
For example, file [61 EF BB BF 62 20 63] should be presented as:
<pre>
ab c
</pre>
but
<pre>
ab  c
</pre>
is observed:
![far_bug](https://user-images.githubusercontent.com/18698375/100750902-b87bec80-33f7-11eb-9b02-d8e52b909990.png)

When trying to edit such file, a mess with characters appearing in wrong place happens around this place.

[bug.zip](https://github.com/elfmz/far2l/files/5623041/bug.zip)

**Comment by elfmz** [comment]
BOM in the middle of test. is it legal at all?

**Comment by e673** [comment]
Yes, it is not punishable for files to contain any sequence of bytes. For example, in appears in RFC 5424 - every message follows after text header and is prefixed with BOM mark. So viewing log file is buggy.
